### PR TITLE
JCL Stubs for JEP 359

### PR DESCRIPTION
**Pull request by theresa-m**
Fixes: https://github.com/eclipse/openj9/issues/7973

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins test sanity zlinux jdknext

**Comment by pshipton** [comment]
@AdamBrousseau isn't `jdknext` supposed to work?
```
java.lang.NumberFormatException: For input string: "next"
	at java.lang.NumberFormatException.forInputString(Unknown Source)
	at java.lang.Integer.parseInt(Unknown Source)
	at java.lang.Integer.valueOf(Unknown Source)
	at org.codehaus.groovy.runtime.StringGroovyMethods.toInteger(StringGroovyMethods.java:3312)
	at org.codehaus.groovy.runtime.StringGroovyMethods.asType(StringGroovyMethods.java:171)
```

**Comment by dnakamura** [comment]
@pshipton Oops, this one is my bad. See #7981

**Comment by pshipton** [comment]
New compile failure with these changes.
https://ci.eclipse.org/openj9/job/Build_JDKnext_s390x_linux_Personal/19
```
14:09:52  /home/jenkins/jenkins-agent/workspace/Build_JDKnext_s390x_linux_Personal/build/linux-s390x-server-release/support/j9jcl_sources/java.base/share/classes/java/lang/Class.java:4475: error: cannot find symbol
14:09:52  	public RecordComponent[] getRecordComponents() {
14:09:52  	       ^
14:09:52    symbol:   class RecordComponent
14:09:52    location: class Class<T>
14:09:52    where T is a type-variable:
14:09:52      T extends Object declared in class Class
```

**Comment by pshipton** [comment]
jenkins compile zlinux jdknext

**Comment by pshipton** [comment]
@theresa-m when you fix the latest problem, pls also rebase to resolve the NumberFormatException in PR testing.

**Comment by theresa-m** [comment]
The failing build from #7973 was based on the `openj9-staging` branch from the extensions repo and these pr builds are based on the `openj9` branch which does not contain support for records.

**Comment by pshipton** [comment]
The build in https://github.com/eclipse/openj9/pull/7974#issuecomment-562275018 was supposed to be based on the openj9-staging branch, but I see there is a problem, perhaps in the pipeline code. I've started a new build https://ci.eclipse.org/openj9/job/Build_JDKnext_s390x_linux_Personal/21/

**Comment by theresa-m** [comment]
^ rebase for the NumberFormatException exception

**Comment by pshipton** [comment]
jenkins compile plinux jdk8

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 1]
```diff
@@ -4689,4 +4689,26 @@ public String descriptorString() {
...
+	 * Returns true if the class instance is a record.
+	 * 
+	 * @return true for a record class, false otherwise
+	 */
+	public boolean isRecord() {
+		throw new InternalError("Compile stub invoked! For JEP 359 support see https://github.com/eclipse/openj9/pull/7946"); //$NON-NLS-1$
+	}
+
+	/**
+	 * Returns an array  of RecordComponent objects for a record class.
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/Class.java]: There is an extra space here between the words `array  of`. Pls fix it in #7946

**Comment by pshipton** [comment]
Testing for #7992, I'll be cancelling this build.

jenkins compile plinux jdk11

**Pull request merged**